compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -627,6 +628,20 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
         self.buffer_error(err);
     }
 
+    fn suggest_map_index_mut_alternatives(
+        &self,
+        ty: Ty<'_>,
+        err: &mut DiagnosticBuilder<'_, ErrorGuaranteed>,
+    ) {
+        let Some(adt) = ty.ty_adt_def() else { return };
+        let did = adt.did();
+        if self.infcx.tcx.is_diagnostic_item(sym::HashMap, did)
+            || self.infcx.tcx.is_diagnostic_item(sym::BTreeMap, did)
+        {
+            err.help("you can use `.get_mut()`, `.insert()` or the entry API");
```

Comment:
```suggestion
            err.help("to insert into a {ty}, use `.get_mut()`, `.insert()` or the entry API");
```
nit: I think it might be a little bit clearer to explain what we're assuming the user is intending.

Reply:
But I don't know whether the user intends to insert or change an existing value. I can assume if this is assignment this is intended to insert, but even then this is not obvious and also what in other cases?

Reply:
Maybe this then?

```suggestion
            err.help("to modify a `{ty}`, use `.get_mut()`, `.insert()` or the entry API");
```

Reply:
Done.